app/src/main/java/net/squanchy/support/widget/ImageViewWithForeground.kt

```diff
@@ -0,0 +1,193 @@
+        if (d != null) {
+            foreground = d
+        }
+
+        foregroundInPadding = a.getBoolean(R.styleable.ImageViewWithForeground_foregroundInsidePadding, true)
+
+        a.recycle()
+    }
+
+    override fun foregroundGravity(): Int = gravity
+
+    override fun setForegroundGravity(foregroundGravity: Int) {
+        if (this.gravity == foregroundGravity) {
+            return
+        }
+
+        var adjustedGravity = foregroundGravity
+        if (adjustedGravity and Gravity.RELATIVE_HORIZONTAL_GRAVITY_MASK == 0) {
+            adjustedGravity = adjustedGravity or Gravity.START
+        }
+
+        if (adjustedGravity and Gravity.VERTICAL_GRAVITY_MASK == 0) {
+            adjustedGravity = adjustedGravity or Gravity.TOP
+        }
+
+        this.gravity = adjustedGravity
+
+        if (this.gravity == Gravity.FILL && foregroundDrawable != null) {
+            val padding = Rect()
+            foregroundDrawable!!.getPadding(padding)
+        }
+
+        requestLayout()
+    }
+
+    // We actually want the instance comparison here
+    override fun verifyDrawable(drawable: Drawable): Boolean {
+        return super.verifyDrawable(drawable) || drawable === foregroundDrawable
+    }
+
+    override fun jumpDrawablesToCurrentState() {
+        super.jumpDrawablesToCurrentState()
+        if (foregroundDrawable != null) {
+            foregroundDrawable!!.jumpToCurrentState()
+        }
+    }
+
+    override fun drawableStateChanged() {
+        super.drawableStateChanged()
+        if (foregroundDrawable != null && foregroundDrawable!!.isStateful) {
+            foregroundDrawable!!.state = drawableState
+        }
+    }
+
+    /**
+     * Supply a Drawable that is to be rendered on top of all of the child
+     * views in the frame layout.  Any padding in the Drawable will be taken
+     * into account by ensuring that the children are inset to be placed
+     * inside of the padding area.
+     *
+     * @param drawable The Drawable to be drawn on top of the children.
+     */
+    override fun setForeground(drawable: Drawable) {
+        if (foregroundDrawable === drawable) {
+            return
+        }
+
+        if (foregroundDrawable != null) {
+            foregroundDrawable!!.callback = null
+            unscheduleDrawable(foregroundDrawable)
+        }
+
+        foregroundDrawable = drawable
+
+        drawable.callback = this
+        if (drawable.isStateful) {
+            drawable.state = drawableState
+        }
+        if (gravity == Gravity.FILL) {
+            val padding = Rect()
+            drawable.getPadding(padding)
+        }
+        requestLayout()
+        invalidate()
+    }
+
+    /**
+     * Returns the drawable used as the foreground of this FrameLayout. The
+     * foreground drawable, if non-null, is always drawn on top of the children.
+     *
+     * @return A Drawable or null if no foreground was set.
+     */
+    override fun foreground(): Drawable? = foregroundDrawable
```

Comment:
`foregroundDrawable` should be a property

Reply:
yeah true, but should we remove the interface ViewWithForeground?

Reply:
Let's leave it for now, but it should be ported over there too

Reply:
👍